packages/domain/src/use-cases/publications/CreatePost.ts

```diff
@@ -11,6 +11,7 @@ import { ReferencePolicyConfig } from './ReferencePolicyConfig';
 import { CollectPolicyConfig, ContentFocus, Locale, Media } from './types';
 
 export type CreatePostRequest = {
+  appId?: AppId;
```

Comment:
I didn't think of passing this as part of the request models, I though of this as an implementation detail. But ultimately `AppId` is an domain concept and the more I look at it the more it's growing on me. Good call!

To fully support this you want to also make sure that when `SupportedTransactionRequest` (of which both `CreatePostRequest` and `CreateCommentRequest` belongs to), you need to make sure it can be serialized and deserialized from storage when part of a "pending transaction".

It boils down to adding zod validation here: https://github.com/lens-protocol/lens-sdk/blob/main/packages/react/src/transactions/adapters/PendingTransactionGateway/schema/publications.ts#L93 and https://github.com/lens-protocol/lens-sdk/blob/main/packages/react/src/transactions/adapters/PendingTransactionGateway/schema/publications.ts#L105

Reply:
it doesn't work for me, how to add AppId type with zod?

This is complaining that a string is not the same type as AppId

```
appId: z.string().optional(),
```

Reply:
This is probably cause `AppId` is branded. It might require a TS assertion or something.

Reply:
This long article explore zod with a library that makes use of branded types: https://www.sandromaglione.com/techblog/zod-and-newtype-ts-full-type-safety-with-typescript

**TL;DR;**
In our case `.transform(appId)` should be enough (not sure if before or after `optional`, you will need to try).

Reply:
I had to replace our implementation of Branded with zod one, to make them compatible, I hope that's fine

```
const AppId = z.string().brand<'AppId'>();
export type AppId = z.infer<typeof AppId>;
```

Reply:
Unfortunately this dependency https://github.com/lens-protocol/lens-sdk/pull/228/files#diff-20cd445a0f23c7bcac95f9d29cdc1bfdb772a0642e6d190367e0d2b0e288dfeaR64

is problematic in terms for dependency rule. Now Domain depends on a 3rd party library.

Reply:
yeah and the build is broken, I can't figure out how to solve it so I guess I will revert adding `zod` to the `domain`

some explanation: https://github.com/microsoft/TypeScript/issues/29808#issuecomment-1320713018

Reply:
ok, solved with `appId: z.string().transform(appId).optional()`